app/models/user.rb

```diff
@@ -559,8 +570,8 @@ def scim_emails
   end
 
   def scim_emails=(emails)
-    email = (emails.find { |email| email.primary == true }) ||
-            (emails.find { |email| email.type == "work" }) ||
+    email = emails.find { |email| email.primary == true } ||
+            emails.find { |email| email.type == "work" } ||
             emails.min
```

Comment:
nitpick, but this could be simplified:

```suggestion
    email = emails.min_by { [it.primary ? 0 : 1, it.type == "work" ? 0 : 1, it] }
```

Or you could implement Comparable in `ScimEmail` and then it's a simple:

```suggestion
    email = emails.min
```

```ruby
ScimEmail = Struct.new("ScimEmail", :value, :primary, :type) do
  include Comparable

  def <=>(other)
    return nil unless other.is_a?(ScimEmail)

    [!primary, type.to_s] <=> [!other.primary, other.type.to_s]
  end
end
```